src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -249,7 +256,7 @@ class RegularSync(
       val blocks = headersQueue.zip(m).map{ case (header, body) => Block(header, body) }
 
       @tailrec
-      def importBlocks(blocks: List[Block], importedBlocks: List[Block] = Nil): (List[Block], Option[BlockImportFailed]) =
+      def importBlocks(blocks: List[Block], importedBlocks: List[Block] = Nil): (List[Block], Option[Any]) =
```

Comment:
why `Any`?

Reply:
with that, the matches above are not exhaustive, doesn't it produce a warning?

Reply:
Could be `BlockImportResult` instead of any... The thing is `UnknownParent` isn't a `BlockImportFailure` - I don't think of it as such because we theoretically still have an option of accepting orphaned blocks (see description).

There is no warning, what matches do you mean?